sale_operating_unit/views/sale_view.xml

```diff
@@ -39,7 +39,23 @@
                     name="operating_unit_id"
                     widget="selection"
                     groups="operating_unit.group_multi_operating_unit"
+                    readonly="state not in ['draft', 'sent']"
                 />
+                <field
+                    name="operating_unit_id"
+                    invisible="1"
+                    groups="!operating_unit.group_multi_operating_unit"
+                />
+            </field>
+            <field name="analytic_account_id" position="attributes">
+                <attribute
+                    name="domain"
+                >["|",("operating_unit_ids","in", operating_unit_id),("operating_unit_ids","in", False)]</attribute>
+            </field>
+            <field name="journal_id" position="attributes">
+                <attribute
+                    name="domain"
+                >["|",("operating_unit_id","=", operating_unit_id),("operating_unit_id","=", False)]</attribute>
```

Comment:
You need to keep the domain define in module `sale` as well: `('type', '=', 'sale')`

Reply:
Fixed 👍